awx/ui/test/e2e/tests/test-jobs-portal-list-actions.js

```diff
@@ -29,33 +28,30 @@ module.exports = {
         const allJobsButton = '#all-jobs-btn';
         const relaunch = `#job-${data.job.id} i[class*="launch"]`;
         const search = '#portal-container-jobs smart-search input';
-        const spinny = 'div.spinny';
 
         portal.load();
 
-        portal.expect.element(allJobsButton).visible;
+        portal.waitForElementVisible(allJobsButton);
         portal.expect.element(allJobsButton).enabled;
         portal.click(allJobsButton);
+        portal.waitForSpinny();
+        portal.assert.cssClassPresent(allJobsButton, 'btn-primary');
 
-        portal.waitForElementVisible(spinny);
-        portal.waitForElementNotVisible(spinny);
 
         const query = `id:>${data.job.id - 1} id:<${data.job.id + 1}`;
 
-        portal.expect.element(search).visible;
+        portal.waitForElementVisible(search);
         portal.expect.element(search).enabled;
         portal.sendKeys(search, query);
         portal.sendKeys(search, client.Keys.ENTER);
 
-        portal.waitForElementVisible(spinny);
-        portal.waitForElementNotVisible(spinny);
+        portal.waitForSpinny();
 
-        portal.expect.element(relaunch).visible;
+        portal.waitForElementVisible(relaunch);
```

Comment:
are these functionally different at all? they seem to do the same thing but also I like matching the syntax used by the existing test, whether it's client.expect(xyz) versus typical selenium calls.

Reply:
@dsesami I am using the wait functions to have some extra wait if needed. It was failing before because the page was not really fully loaded and nightwatch was performing other actions.